core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -43,6 +44,7 @@ class MetadataRequestTest extends BaseRequestTest {
     doSetup(createOffsetsTopic = false)
   }
 
+  //FIX ME
   @Test
   def testClusterIdWithRequestVersion1() {
     val v1MetadataResponse = sendMetadataRequest(MetadataRequest.Builder.allTopics.build(1.toShort))
```

Comment:
@cmccabe Currently automatic protocol generator doesn't support null value as default. For MetadataRequest we need to return null clusterId  for V1 requests. Any thoughts on handling this?

```
kafka.server.MetadataRequestTest > testClusterIdWithRequestVersion1 FAILED
    java.lang.AssertionError: v1 clusterId should be null expected null, but was:<>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotNull(Assert.java:756)
        at org.junit.Assert.assertNull(Assert.java:738)
        at kafka.server.MetadataRequestTest.testClusterIdWithRequestVersion1(MetadataRequestTest.scala:51)
```
cc @rajinisivaram  @ijuma

Reply:
Hi @omkreddy, that's a good point. I think we should allow null default values.  I have a PR here which implements that here: https://github.com/apache/kafka/pull/6387